[freeimage] link failure

**Host Environment**
- OS: Ubuntu 20.04 LTS (WSL)
- Compiler: GNU 9.3.0

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install freeimage`
Create CMakeLists.txt:
```cmake
cmake_minimum_required(VERSION 3.13)
project (FreeImageTest)

add_library(Test SHARED "")

find_package(freeimage CONFIG REQUIRED)
target_link_libraries(Test PRIVATE freeimage::FreeImage)
target_sources(Test PRIVATE Main.cpp)
```
Main.cpp:
```cpp
#include <FreeImage.h>

void foo()
{
	FreeImage_Initialise(0);
}
```
Configure and build using cmake.
**Failure logs** 
ld failed during linking stage:
```
/usr/bin/ld: /mnt/d/TheReclif/Linux/vcpkg/installed/x64-linux/lib/libtiff.a(tif_unix.c.o):(.data.rel.local+0x0): multiple definition of `_TIFFerrorHandler'; /mnt/d/TheReclif/Linux/vcpkg/installed/x64-linux/debug/lib/libFreeImaged.a(PluginTIFF.cpp.o):/mnt/d/TheReclif/Linux/vcpkg/buildtrees/freeimage/src/9d9cc7e2d5-2288df34d2.clean/Source/FreeImage/PluginTIFF.cpp:276: first defined here
/usr/bin/ld: /mnt/d/TheReclif/Linux/vcpkg/installed/x64-linux/lib/libtiff.a(tif_unix.c.o):(.data.rel.local+0x8): multiple definition of `_TIFFwarningHandler'; /mnt/d/TheReclif/Linux/vcpkg/installed/x64-linux/debug/lib/libFreeImaged.a(PluginTIFF.cpp.o):/mnt/d/TheReclif/Linux/vcpkg/buildtrees/freeimage/src/9d9cc7e2d5-2288df34d2.clean/Source/FreeImage/PluginTIFF.cpp:261: first defined here
```
**Additional context**
The problem is caused by linking to the FreeImage and libTIFF libraries at the same time and renders FreeImage unusable. Specifically, it's caused by FreeImage redefining libTIFF's warning and error handlers (lines 261 and 276 in PluginTIFF.cpp, FreeImage's source code). It can be temporarily fixed by removing link commands for libtiff.a in generated makefiles. Proposed solution: don't propagate the libTIFF link dependency from FreeImage's cmake file if that's possible.

--- a/ports/freeimage/CONTROL
+++ b/ports/freeimage/CONTROL
@@ -1,6 +1,6 @@
 Source: freeimage
 Version: 3.18.0
-Port-Version: 15
+Port-Version: 16
 Build-Depends: zlib, libpng, libjpeg-turbo, tiff, openjpeg, libwebp (!uwp), libraw, jxrlib, openexr
 Homepage: https://sourceforge.net/projects/freeimage/
 Description: Support library for graphics image formats

--- a/ports/freeimage/portfile.cmake
+++ b/ports/freeimage/portfile.cmake
@@ -17,6 +17,7 @@ vcpkg_from_sourceforge(
         use-freeimage-config-include.patch
         fix-function-overload.patch
         use-typedef-as-already-declared.patch
+        use-functions-to-override-libtiff-warning-error-handlers.patch
 )
 
 file(COPY ${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt DESTINATION ${SOURCE_PATH})

--- a/ports/freeimage/use-functions-to-override-libtiff-warning-error-handlers.patch
+++ b/ports/freeimage/use-functions-to-override-libtiff-warning-error-handlers.patch
@@ -1,0 +1,31 @@
+diff --git a/Source/FreeImage/PluginTIFF.cpp b/Source/FreeImage/PluginTIFF.cpp
+index a9fcf0e..b1f7b32 100644
+--- a/Source/FreeImage/PluginTIFF.cpp
++++ b/Source/FreeImage/PluginTIFF.cpp
+@@ -258,7 +258,7 @@ static void
+ msdosWarningHandler(const char* module, const char* fmt, va_list ap) {
+ }
+ 
+-TIFFErrorHandler _TIFFwarningHandler = msdosWarningHandler;
++//TIFFErrorHandler _TIFFwarningHandler = msdosWarningHandler;
+ 
+ static void
+ msdosErrorHandler(const char* module, const char* fmt, va_list ap) {
+@@ -273,7 +273,7 @@ msdosErrorHandler(const char* module, const char* fmt, va_list ap) {
+ 	*/
+ }
+ 
+-TIFFErrorHandler _TIFFerrorHandler = msdosErrorHandler;
++//TIFFErrorHandler _TIFFerrorHandler = msdosErrorHandler;
+ 
+ // ----------------------------------------------------------
+ 
+@@ -2659,6 +2659,8 @@ InitTIFF(Plugin *plugin, int format_id) {
+     // Set up the callback for extended TIFF directory tag support (see XTIFF.cpp)
+ 	// Must be called before using libtiff
+     XTIFFInitialize();	
++	TIFFSetWarningHandler(msdosWarningHandler);
++	TIFFSetErrorHandler(msdosErrorHandler);
+ 
+ 	plugin->format_proc = Format;
+ 	plugin->description_proc = Description;

